src/server/graph-generation/interaction/index.js

```diff
@@ -56,20 +64,85 @@ function parse(data, queryIds){
     edges:[],
     nodes:[],
   };
-  let nodeMap=new Map(); //keeps track of nodes that have already been added
+  let nodeDegreeMap =new Map(); //keeps track of nodes that have already been added, also calculates node degree
```

Comment:
What about just having an id => node map?  You can just use node.data.metric directly